xcodeproj/internal/bazel_integration_files/swiftc.swift

```diff
@@ -0,0 +1,115 @@
+}
+
+/// Touch the Xcode-required .d files
+func touchDepsFiles(_ args: [String]) throws {
+    guard let outputFileMapPath = findPath(key: "-output-file-map", from: args)
+    else { return }
+
+    if isWMO(args: args) {
+        let dFile = String(outputFileMapPath.path.dropLast("-OutputFileMap.json".count) + "-master.d")
+        var url = URL(fileURLWithPath: dFile)
+        try url.touch()
+    } else {
+        var swiftFileListPath: String?
+        for arg in args {
+            if arg.hasPrefix("@") {
+                swiftFileListPath = String(arg.dropFirst())
+                break
+            }
+        }
+
+        let parentDir = outputFileMapPath.deletingLastPathComponent()
+        let swiftFilenames = try String(contentsOfFile: swiftFileListPath!, encoding: .utf8)
+            .components(separatedBy: .newlines)
+            .compactMap { $0.components(separatedBy: "/").last }
+
+        for filename in swiftFilenames {
+            let dFileBasename = filename.dropLast(".swift".count) + ".d" as String
+            var url = URL(fileURLWithPath: dFileBasename, relativeTo: parentDir)
+            try url.touch()
+        }
+    }
+}
+
+/// Touch the Xcode-required .swift{module,doc,sourceinfo} files"
+func touchSwiftmoduleArtifacts(_ args: [String]) throws {
+    if var swiftmodulePath = findPath(key: "-emit-module-path", from: args) {
+        var swiftdocPath = swiftmodulePath.deletingPathExtension().appendingPathExtension("swiftdoc")
+        var swiftsourceinfoPath = swiftmodulePath.deletingPathExtension().appendingPathExtension("swiftsourceinfo")
+        var swiftinterfacePath = swiftmodulePath.deletingPathExtension().appendingPathExtension("swiftinterface")
+
+        try swiftmodulePath.touch()
+        try swiftdocPath.touch()
+        try swiftsourceinfoPath.touch()
+        try swiftinterfacePath.touch()
+    }
+
+    if var generatedHeaderPath = findPath(key: "-emit-objc-header-path", from: args) {
+        try generatedHeaderPath.touch()
+    }
+}
+
+func runSubProcess(args: [String]) throws -> Int32 {
+    let task = Process()
+    task.launchPath = args.first
+    task.arguments = Array(args.dropFirst())
+    try task.run()
+    task.waitUntilExit()
+    return task.terminationStatus
+}
+
+// MARK: - Main
+
+let args = CommandLine.arguments
+
+if args.count == 1, args.last == "-v" {
+    exit(try runSubProcess(args: ["swiftc", "-v"]))
+}
+
+for arg in args {
+    // Pass through for SwiftUI Preview thunk compilation
+    if arg.hasSuffix(".preview-thunk.swift"), args.contains("-output-file-map") {
+        guard let sdkPath = findPath(key: "-sdk", from: args)?.path else { break }
+        let fullRange = NSRange(sdkPath.startIndex..., in: sdkPath)
+        let matches = try NSRegularExpression(pattern: #".*?/Contents/Developer)/.*"#)
+            .matches(in: sdkPath, range: fullRange)
+        guard let developerDir = matches.first else { break }
```

Comment:
Both of these error states should result in a warning and then use plain `swiftc` instead of falling back to normal argument parsing. After we've seen that this should be for SwiftUI Previews, we need to call the underlying `swiftc`.